pkg/apis/workloads/v1alpha1/console_types.go

```diff
@@ -93,6 +101,15 @@ type ConsoleTemplateSpec struct {
 	// +kubebuilder:validation:Maximum=604800
 	MaxTimeoutSeconds        int              `json:"maxTimeoutSeconds"`
 	AdditionalAttachSubjects []rbacv1.Subject `json:"additionalAttachSubjects,omitempty"`
+
+	// Specifies the TTL for any Console created with this template. If set, the Console
+	// will be eligible for garbage collection ConsoleTTLSecondsAfterFinished seconds after
+	// it enters the Stopped phase. If not set, this value defaults to 24 hours.
+	// This field is modeled closely on the TTL mechanism in Kubernetes 1.12.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=604800
+	TTLSecondsAfterFinished *int32 `json:"ttlSecondsAfterFinished,omitempty"`
```

Comment:
I've gone for `*int32` here because it matches the type of the field of the same name on Jobs in Kube 1.12